sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DisposableReturnedFromUsing.cs

```diff
@@ -57,47 +58,67 @@ protected override void Initialize(SonarAnalysisContext context)
-                            {
-                                return;
-                            }
-                            declaredSymbols = new HashSet<ISymbol> { symbol };
+                            return;
                         }
+                        var symbol = c.SemanticModel.GetSymbolInfo(identifierName).Symbol;
+                        if (symbol == null)
+                        {
+                            return;
+                        }
+                        declaredSymbols = new HashSet<ISymbol> { symbol };
                     }
 
-                    if (declaredSymbols.Count == 0)
+                    CheckReturns(c, usingStatement.UsingKeyword, usingStatement.Statement, declaredSymbols);
+                },
+                SyntaxKind.UsingStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var localDeclarationStatement = (LocalDeclarationStatementSyntax)c.Node;
+                    var usingKeyword = localDeclarationStatement.UsingKeyword();
+                    if (!usingKeyword.IsKind(SyntaxKind.UsingKeyword))
                     {
                         return;
                     }
 
-                    var returnedSymbols = GetReturnedSymbols(usingStatement.Statement, c.SemanticModel);
-                    returnedSymbols.IntersectWith(declaredSymbols);
+                    var declaredSymbols = localDeclarationStatement.Declaration.Variables
+                                .Select(syntax => c.SemanticModel.GetDeclaredSymbol(syntax))
+                                .WhereNotNull()
+                                .ToHashSet();
 
-                    if (returnedSymbols.Any())
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, usingStatement.UsingKeyword.GetLocation(),
-                            string.Join(", ",
-                                returnedSymbols.Select(s => $"'{s.Name}'").OrderBy(s => s))));
-                    }
+                    CheckReturns(c, usingKeyword, localDeclarationStatement.Parent, declaredSymbols);
                 },
-                SyntaxKind.UsingStatement);
+                SyntaxKind.LocalDeclarationStatement);
+        }
+
+        private static void CheckReturns(SyntaxNodeAnalysisContext c, SyntaxToken usingKeyword, SyntaxNode body, HashSet<ISymbol> declaredSymbols)
+        {
+            if (declaredSymbols.Count == 0)
+            {
+                return;
+            }
+
+            var returnedSymbols = GetReturnedSymbols(body, c.SemanticModel);
+            returnedSymbols.IntersectWith(declaredSymbols);
+
+            if (returnedSymbols.Any())
+            {
+                c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, usingKeyword.GetLocation(),
+                    string.Join(", ",
+                        returnedSymbols.Select(s => $"'{s.Name}'").OrderBy(s => s))));
```

Comment:
could you add a test where this concatenation is used? like more symbols in the issue message?